Periodic_3_mesh_3/include/CGAL/refine_periodic_3_mesh_3.h

```diff
@@ -133,52 +133,237 @@ void project_points(C3T3& c3t3,
-      (lloyd_param, (parameters::internal::Lloyd_options), parameters::no_lloyd())
-      (reset_param, (parameters::Reset), parameters::reset_c3t3())
-      (mesh_options_param, (parameters::internal::Mesh_3_options),
-       parameters::internal::Mesh_3_options())
-      (manifold_options_param, (parameters::internal::Manifold_options),
-       parameters::internal::Manifold_options())
-    )
-  )
-)
+template<typename C3T3, typename MeshDomain, typename MeshCriteria, typename CGAL_NP_TEMPLATE_PARAMETERS>
+void refine_periodic_3_mesh_3(C3T3& c3t3, MeshDomain& domain, MeshCriteria& criteria, const CGAL_NP_CLASS& np = parameters::default_values())
 {
-  return refine_periodic_3_mesh_3_impl(c3t3, domain, criteria,
-                                       exude_param,
-                                       perturb_param,
-                                       odt_param,
-                                       lloyd_param,
-                                       reset_param(),
-                                       mesh_options_param,
-                                       manifold_options_param);
+    using parameters::choose_parameter;
+    using parameters::get_parameter;
+    parameters::internal::Exude_options exude_param = choose_parameter(get_parameter(np, internal_np::exude_options_param), parameters::no_exude());
+    parameters::internal::Perturb_options perturb_param = choose_parameter(get_parameter(np, internal_np::perturb_options_param), parameters::no_perturb());
+    parameters::internal::Odt_options odt_param = choose_parameter(get_parameter(np, internal_np::odt_options_param), parameters::no_odt());
+    parameters::internal::Lloyd_options lloyd_param = choose_parameter(get_parameter(np, internal_np::lloyd_options_param), parameters::no_lloyd());
+    parameters::Reset reset_param = choose_parameter(get_parameter(np, internal_np::reset_options_param), parameters::reset_c3t3());
+    parameters::internal::Mesh_3_options mesh_options_param = choose_parameter(get_parameter(np, internal_np::mesh_param), parameters::internal::Mesh_3_options());
+    parameters::internal::Manifold_options manifold_options_param = choose_parameter(get_parameter(np, internal_np::manifold_param), parameters::internal::Manifold_options());
+
+    return refine_periodic_3_mesh_3_impl(c3t3,
+                              domain,
+                              criteria,
+                              exude_param,
+                              perturb_param,
+                              odt_param,
+                              lloyd_param,
+                              reset_param(),
+                              mesh_options_param,
+                              manifold_options_param);
 }
 
-CGAL_PRAGMA_DIAG_POP
-
-#if defined(BOOST_MSVC)
-#  pragma warning(pop)
-#endif
+    template<typename CGAL_NP_TEMPLATE_PARAMETERS_NO_DEFAULT>
+    void refine_periodic_3_mesh_3(const CGAL_NP_CLASS& np)
```

Comment:
This is not a good idea: With function named parameters, we were only adding optional named parameters. I understand that with replacing Boost Named Parameter, we have to allow some optional parameter to be mandatory but I would not add new mandatory named parameter like c3t3. Especially because here this call is going to give nothing as output to the user. The c3t3 will be copied, the refinement being done on the copy and the output will be the input.